scripts/data/client.py

```diff
@@ -12,36 +11,45 @@
 from concurrent.futures import ThreadPoolExecutor
 from pathlib import Path
 import random
+import signal
+import sys
 from generate_data import generate_data
 from format_args import format_args
-import logging
 from logging.handlers import TimedRotatingFileHandler
 
+# Initialize logger
 logger = logging.getLogger(__name__)
 
 # Constants
 MAX_WEIGHT_LIMIT = 8000  # Total weight limit for all jobs
 THREAD_POOL_SIZE = os.cpu_count()  # Number of threads for processing
 QUEUE_MAX_SIZE = THREAD_POOL_SIZE * 2  # Maximum size of the job queue
-
-BASE_DIR = Path(".client_cache")
+BASE_DIR = Path(__file__).resolve().parent / ".client_cache"  # Use absolute path
```

Comment:
Why this change is necessary?

Reply:
Although I've reverted the change, but using an absolute path for `BASE_DIR` is better than a relative path because it ensures consistent behaviour regardless of where the script is executed from.